pkg/check/peercount.go

```diff
@@ -25,7 +26,13 @@ func PeerCount(opts PeerCountOptions) (err error) {
 	var expectedPeerCount = opts.NodeCount + opts.BootNodeCount - 1
 
 	for i := 0; i < opts.NodeCount; i++ {
-		nodeURL, err := url.Parse(fmt.Sprintf(debugAPITemplateURL, i, opts.Namespace))
+		var nodeURL *url.URL
+		var err error
+		if opts.URLTemplate != "" {
+			nodeURL, err = url.Parse(fmt.Sprintf(opts.URLTemplate, i))
```

Comment:
This Sprintf is assuming that the url template will not have a namespace in it. I am suggesting to use template package to format the url instead Sprintf. This can be done in a separate PR as it extends its scope, but this implementation should not stay for long.